feat(RELEASE-1387): Use trusted artifacts with stepactions

scoheb opened the pull request:
## Describe your changes
- in an effort to move away from PVCs, we are trying
  trusted artifacts based on https://github.com/konflux-ci/build-trusted-artifacts/tree/main
- tasks that have been updated now support 2 modes:
  - PVC based workspaces
  - Trusted artifacts with emptyDir workspaces
    
Summary:
- StepActions that support trusted artifacts have been added.
- all tasks used in the rh-advisory pipeline have been converted to
  support trusted artifacts mode
- pipelines have been updated to include the stepAction git related
  parameters.
- pipelines that use apply-mapping have had the workspace used for
  the apply-mapping task renamed to be in line with other tasks.
- The task verify-enterprise-contract has been renamed to
  verify-conforma and has been refactored to use the new git-resolved
  task verify-conforma-konflux-ta
  
- CI changes:
  - CI now attempts to detect if a Trusted Artifacts based task
     has been changed and runs the tests with trusted artifacts enabled.
  - tests now create an on-local-cluster registry.
  - CI will find stepactions and install them on local cluster
    and update git resolvers to be direct cluster resolvers
    
requires https://github.com/konflux-ci/release-service/pull/666
for use on real clusters

**This supercedes #784** 

## Relevant Jira
- [RELEASE-1387](https://issues.redhat.com//browse/RELEASE-1387)

## Checklist before requesting a review
- [X] I have marked as draft or added `do not merge` label if there's a dependency PR
  - If you want reviews on your draft PR, you can add reviewers or add the `release-service-maintainers` handle if you are unsure who to tag
- [X] My commit message includes `Signed-off-by: My name <email>`
- [X] I have bumped the task/pipeline version string and updated changelog in the relevant README
- [X] I read CONTRIBUTING.MD and [commit formatting](CONTRIBUTING.md#commit-message-formatting-and-standards)

scoheb committed:
feat(RELEASE-1387): use trusted artifacts

- in an effort to move away from PVCs, we are trying
  trusted artifacts based on
  https://github.com/konflux-ci/build-trusted-artifacts/tree/main
- tasks that have been updated now support 2 modes:
  - PVC based workspaces
  - Trusted artifacts with emptyDir workspaces

Summary:
- StepActions that support trusted artifacts have been added.
- all tasks used in the rh-advisory pipeline have been converted to
  support trusted artifacts mode
- pipelines have been updated to include the stepAction git related
  parameters.
- pipelines that use apply-mapping have had the workspace used for
  the apply-mapping task renamed to be in line with other tasks.
- The task verify-enterprise-contract has been renamed to
  verify-conforma and has been refactored to use the new git-resolved
  task verify-conforma-konflux-ta

- CI changes:
  - CI now attempts to detect if a Trusted Artifacts based task
     has been changed and runs the tests with trusted artifacts enabled.
  - tests now create an on-local-cluster registry.
  - CI will find stepactions and install them on local cluster
    and update git resolvers to be direct cluster resolvers

requires https://github.com/konflux-ci/release-service/pull/666
for use on real clusters

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb commented:
/test release-service-catalog

scoheb commented:
/retest

scoheb commented:
/retest

scoheb commented:
/retest

scoheb commented:
/retest

scoheb commented:
/retest

johnbieren commented on .github/scripts/deploy_registry.sh:
```diff
@@ -0,0 +1,69 @@
+
+prepare_docker_config() {
+  authString=$(echo -n "root:root" | base64 -w0)
+  cat > "/tmp/.dockerconfig.json" <<EOF
```
Copying my thread from the other PR. It was left as
```
I think the idea would be that deploy_registry.sh would create a temp file and export the value as DOCKER_CONFIG. And then test_tekton_tasks.sh would consume it. I'm fine with the current approach though.
```
from Martin. The reason why I asked for `mktemp` is for when you run this locally

scoheb replied:
I don't see why you are concerned about this code when you run this locally?

theflockers replied:
I think by using mktemp you make sure the config you are generating is unique for that run - so no leftovers would cause instability in the script (for whatever reasons) - also imagine that I can (also, for any reason) use /tmp, you would just  `export TEMP=something-else`  and your temp would exist elsewhere.

I am just giving some reasons why it would be a nice to have, but I am fine with not changing as well.

scoheb replied:
Can we agree to change this if need be in a future PR?

johnbieren replied:
I mean there is no shot this ever gets changed once this PR is merged. Is `/tmp/.dockerconfig.json` not the most common spot you'd put a docker config if you were doing something locally? I wouldn't want to overwrite that. It is just my thoughts on it.

scoheb replied:
first ... `/tmp/.dockerconfig.json` is not a known accepted, common path for a docker config. We can name it something more specific if we want.

second ... `deploy-registry.sh` is meant to be executed to setup your environment. Then many subsequent runs of `test_tekton_tasks.sh` can take place. How would `deploy_registry.sh` be able to set an environment variable to be available for `test_tekton_tasks.sh`?

consider this:

`deploy-registry.sh`

```
#!/usr/bin/env bash

dockerconfigFile="$(mktemp -d)/.dockerconfig.json"
export DOCKER_CONFIG_JSON=$dockerconfigFile
```

`test_tekton_tasks.sh`

```
#!/usr/bin/env bash

echo "DOCKER_CONFIG_JSON: $DOCKER_CONFIG_JSON"
```

Running this leads to:

```
% deploy-registry.sh
% test_tekton_tasks.sh
DOCKER_CONFIG_JSON: 
```

am I missing something?

scoheb replied:
ok...so I think I have it working. Please see the updates.

essentially, `deploy-registry.sh` must be sourced...

- `deploy-registry.sh` now has a check to ensure it is sourced.
- updated the workflow to call `. deploy-registry.sh`
- update CONTRIBUTING.md to call this out.

johnbieren replied:
love it :wink:

johnbieren commented on .github/scripts/test_tekton_tasks.sh:
```diff
@@ -116,7 +181,15 @@ do
     done
 
-    PIPELINERUN=$(tkn p start $TEST_NAME -w name=tests-workspace,volumeClaimTemplateFile=$WORKSPACE_TEMPLATE -o json | jq -r '.metadata.name')
+    if [ -z "${USE_TRUSTED_ARTIFACTS}" ]; then
```
aren't these already set on lines 147-153?

scoheb replied:
nice catch. fixed

johnbieren commented on find-non-ta-tasks.sh:
Not entirely sure what this is used for, but I don't think it should be in the root of the repository

scoheb replied:
moved

johnbieren commented on find-ta-tasks.sh:
Not entirely sure what this is used for, but I don't think it should be in the root of the repository

johnbieren replied:
Also, this is almost identical to find-non-ta-tasks.sh. Can the scripts be combined into one? It could take an argument to switch between the two

scoheb replied:
merged to a single file and moved to scripts folder

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
```diff
@@ -59,6 +59,21 @@ spec:
     - name: taskGitRevision
       type: string
       description: The revision in the taskGitUrl repo to be used
+    - name: ociStorage
```
These are all missing from the README

scoheb replied:
added

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
```diff
@@ -191,31 +227,47 @@ spec:
+          - name: url
+            value: https://github.com/enterprise-contract/ec-cli
+          - name: revision
+            value: main
```
we can use main? we had to use a value from the configmap before to ensure it was a known good version. latest was not okay for us to use

scoheb replied:
Opened a slack thread to discuss: https://redhat-internal.slack.com/archives/C031J4KBFME/p1743456151444029

scoheb replied:
ec cli git sha param now passed from controller via ec configmap

johnbieren replied:
I see this now in rh-advisories.yaml
```
    - name: verify-enterprise-contract
      timeout: "4h00m0s"
      taskRef:
        resolver: "bundles"
        params:
          - name: bundle
            value: $(params.verify_ec_task_bundle)
          - name: kind
            value: task
          - name: name
            value: verify-enterprise-contract
```
but if the param is passed from the controller and it switches to git resolver (like I think your PR used to do), you'll need to add the param to the pipeline itself

scoheb replied:
`rh-advisories.yaml` as-is will be going away ... to be replaced by the contents of `rh-advisories-oci-ta.yaml`

johnbieren commented on stepactions/create-trusted-artifact/create-trusted-artifact.yaml:
```diff
@@ -0,0 +1,28 @@
+---
+apiVersion: tekton.dev/v1beta1
```
v1 please

scoheb replied:
fixed

scoheb replied:
only exists as **v1beta1** ... see https://github.com/tektoncd/pipeline/tree/main/pkg/apis/pipeline/v1beta1

johnbieren commented on stepactions/patch-source-data-artifact-result/patch-source-data-artifact-result.yaml:
```diff
@@ -0,0 +1,32 @@
+---
+apiVersion: tekton.dev/v1beta1
```
v1 please

scoheb replied:
fixed

scoheb replied:
only exists as **v1beta1** ... see https://github.com/tektoncd/pipeline/tree/main/pkg/apis/pipeline/v1beta1

johnbieren commented on stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations.yaml:
```diff
@@ -0,0 +1,37 @@
+---
+apiVersion: tekton.dev/v1beta1
```
v1 please

scoheb replied:
fixed

scoheb replied:
only exists as **v1beta1** ... see https://github.com/tektoncd/pipeline/tree/main/pkg/apis/pipeline/v1beta1

johnbieren commented on stepactions/use-trusted-artifact/use-trusted-artifact.yaml:
```diff
@@ -0,0 +1,22 @@
+---
+apiVersion: tekton.dev/v1beta1
```
v1 please

scoheb replied:
fixed

scoheb replied:
only exists as **v1beta1** ... see https://github.com/tektoncd/pipeline/tree/main/pkg/apis/pipeline/v1beta1

scoheb committed:
feat(RELEASE-1387): use trusted artifacts - fixes

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb commented:
/retest

scoheb committed:
feat(RELEASE-1387): use trusted artifacts - rename

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): use trusted artifacts - reduce

- collect-data step and result count makes result collection
  and reporting in Tekton brittle.
- reduce the number of steps as this should help
- the issue we are trying to solve is termination messages
  being incomplete.

Signed-off-by: Scott Hebert <scoheb@gmail.com>

jinqi7 commented:
/retest

scoheb committed:
feat(RELEASE-1387): rh-push-to-registry-redhat-io

- use trusted artifacts in rh-push-to-registry-redhat-io

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb commented:
including Jing's change for debugging of e2e tests

/retest

scoheb commented:
/retest

scoheb commented:
/retest

jinqi7 commented:
/retest

jinqi7 commented:
/retest

jinqi7 commented:
/retest

theflockers commented on .github/scripts/find-ta-tasks.sh:
```diff
@@ -0,0 +1,30 @@
+  fi
+done
+
+echo ""
```
why not a single 
```
echo "
BLA

BLA"
```

scoheb replied:
sure. I can change if you feel strongly about it.

theflockers replied:
IMHO, this fells like a "dripping tap". For a single case, it does not represent much, but a full cluster wasting resources repeating commands that consumes (even a bit) of cpu clocks/pids/fd  for  something that could be done once.. not good - again, IMHO - if you don't feel like changing it, NP.

scoheb replied:
this script is actually meant to be run on your laptop...as a means to see which tasks have not been converted yet. It is NOT run as part of a pipeline.

theflockers commented on .github/workflows/tekton_task_tests.yaml:
```diff
@@ -55,14 +82,32 @@ jobs:
           kubectl get pods --namespace tekton-pipelines
+
+          # let tekton controllers startup...
+          sleep 10
```
Some suggestion :

```
timeout 30  bash -c 'while ! kubectl get pod -n tekton-pipelines -l "app=tekton-pipelines-controller" 
--field-selector=status.phase==Running 
|grep tekton; do echo "Waiting for tekton..."; sleep 1; done '
```
It seems too much but it has the advantage of waiting the proper time tekton would take to bring up - up to 30 seconds. Useful if for any reason the load is too high and it would take longer than expected 10 seconds, but not more than 30 (or maybe more).

scoheb replied:
by this time, Tekton is running but the feature flags configMap may not have been created. I can add a comment about that.

theflockers commented on pipelines/managed/push-disk-images-to-cdn/README.md:
```diff
@@ -20,6 +20,12 @@ Tekton Pipeline to push disk images to a cdn using pulp
 | taskGitRevision                 | The revision in the taskGitUrl repo to be used                                                         | No       | -               |
 
+## Changes in 1.6.0
+* Update all tasks that now support trusted artifacts to specify the stepAction* parameters
```
Is the single asterisk in _stepAction_ intentional? Just wondering it you wanted somehow to highlight, add a footnote or if it was just a typo. Same for all the other occurences.

scoheb replied:
intentional. there are 2 params.

theflockers replied:
Maybe changing it to `stepAction(A|B)` - regex style. Maybe I am  the only one that got confused by it, although ater you said, it makes sense.  Anyhow, not crucial change. If you think this is not required, feel free just to resolve the conversation.

scoheb replied:
thanks

theflockers commented on tasks/managed/apply-mapping/apply-mapping.yaml:
```diff
@@ -22,22 +22,112 @@ spec:
+      description: Expiration date for the trusted artifacts created in the
+        OCI repository. An empty string means the artifacts do not expire
+      type: string
+      default: "1d"
```
I suppose these artifacts are temporary, correct? Or would them be reused in case of pipeline failing? Asking because if they are not reused somehow, 1d might be too much. WDYT.

scoheb replied:
correct. temporary. I choose `1d` to allow someone to inspect the artifacts *if* (because of timezone) the pipeline happened overnight, etc

theflockers replied:
got it. resolving.

theflockers commented on tasks/managed/apply-mapping/apply-mapping.yaml:
```diff
@@ -22,22 +22,112 @@ spec:
     - name: mapped
       type: string
       description: A true/false value depicting whether or not the snapshot was mapped.
+    - description: Produced trusted data artifact
```
double description

scoheb replied:
it is not a duplicate. the fields are not ordered correctly. I will fix.

scoheb committed:
feat(RELEASE-1387): trusted artifacts - pr fixes

Signed-off-by: Scott Hebert <scoheb@gmail.com>

johnbieren commented on pipelines/managed/push-disk-images-to-cdn/README.md:
```diff
@@ -20,10 +20,17 @@ Tekton Pipeline to push disk images to a cdn using pulp
 
+## Changes in 1.5.0
+* Align workspace name with changes in the apply-mapping task
+>>>>>>> c969a5f8 (feat(RELEASE-1387): use trusted artifacts)
```
rebase error

scoheb replied:
fixed

johnbieren commented on pipelines/managed/push-disk-images-to-cdn/README.md:
```diff
@@ -20,10 +20,17 @@ Tekton Pipeline to push disk images to a cdn using pulp
 * Update the pipeline to add taskGiturl and taskGitRevison parameters to
   push-disk-images task.
 
+## Changes in 1.5.0
```
there is already a 1.5.0 in the changelog

scoheb replied:
fixed

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories-oci-ta.yaml:
Why is this being added? I thought since rh-advisories has every task updated in this PR, we don't need this

scoheb replied:
I needed to be able to have the emptyDir rule in the ReleaseServiceConfig file to be able to run the e2e tests and integration tests on staging.

If I would have kept the filename `rh-advisories.yaml`, that would have required a rule in the Development layer and having that would have broken everyone.

Once this merges, we can coordinate the infra-deployments layers with the renaming of the pipeline files.

johnbieren replied:
So are the two files a copy? Then once we merge and promote we remove the -ta one?

scoheb replied:
> Then once we merge and promote we remove the -ta one?

exactly.

In sync, we update the RSC in development layer and when we promote and remove the -ta one.

johnbieren replied:
I don't get it. A second advisory yaml is being added, which is vastly different
```
$ diff rh-advisories.yaml rh-advisories-oci-ta.yaml | wc -l
333
```
so the original pipeline yaml is being updated, *and* a new one is being added that is different. If we need the -ta one to enable TA in e2e, why is the original one being updated? Also, the e2e test uses `rh-advisories.yaml` so I don't get how the -ta one affects e2e at all

scoheb replied:
so...

> so the original pipeline yaml is being updated

Yes...since the Tasks are changing to support the use of the TA stepActions:

```
        - name: taskGitUrl
          value: "$(params.taskGitUrl)"
        - name: taskGitRevision
          value: "$(params.taskGitRevision)"
```

> Also, the e2e test uses rh-advisories.yaml so I don't get how the -ta one affects e2e at all

I have a paired e2e-tests PR that uses `rh-advisories.yaml`

Once my PR merges, e2e will not be testing `rh-advisories-oci-ta.yaml`

Here's the plan:

1. Merge PR once all reviews are complete and all tests pass (including e2e test using paired e2e-tests PR)
- the e2e-tests paired PR will NOT be merged.
2. At this point, 2 yamls will be present. `rh-advisories.yaml` and `rh-advisories-oci-ta.yaml`. 
3. `rh-advisories.yaml` is the one active in RPAs
4. Prepare infra RSC change to target `rh-advisories.yaml`
5. Prepare release service catalog to rename `rh-advisories-oci-ta.yaml` to `rh-advisories.yaml`
6. Merge infra and release catalog rename PR at same time.
7. The TA version of `rh-advisories.yaml` is the one active in RPAs

The same holds true to `rh-push-to-registry-redhat-io-oci-ta.yaml`

Alternatively, we can skip the rename PR and have the TA version of `rh-advisories.yaml` directly in my PR.
Essentially, once the e2e-tests pass with the paired PR, perform the rename in my PR knowing the e2e test will fail since the RSC in development still points to the `oci-ta.yaml` path.

open to ideas on how to minimize impact and risk.

johnbieren replied:
Okay, so I will restate the point I am getting at to make sure I understand.
The -ta pipeline yamls were to pair with your e2e pr which already passed. The e2e pr isn't going to be merged. No users should use the -ta pipelines. Once this merges, there will be a new PR that updates the rh-advisories.yaml (and other one) to be what the -ta versions of them are (I know you same rename, but github is probably going to show it as changing the current file with a big diff). 
So, it sounds like we don't need to merge the -ta ones to me. This, IMO, makes it simpler as then we can review those changes in the followup PR. With the current setup, it is like I am approving the -ta ones, even though nothing is going to use them (nothing merged at least) and I will still have to review them when the rename is done anyways. That's why I am advocating for dropping them from this PR

johnbieren commented on pipelines/managed/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io-oci-ta.yaml:
same comment as the advisories one

johnbieren commented on tasks/managed/apply-mapping/apply-mapping.yaml:
```diff
@@ -22,22 +22,112 @@ spec:
+        value: "$(params.orasOptions)"
+      - name: "DEBUG"
+        value: "$(params.trustedArtifactsDebug)"
+      - name: "HOME"
```
Is this needed? We already set it https://github.com/konflux-ci/release-service-utils/blob/main/Dockerfile#L86 (applies in many places)

scoheb replied:
I removed it. we'll see if CI complains.

scoheb replied:
CI complained...I need to set `HOME` for the StepActions that do not use the release-service-utils image.

Updated...

johnbieren replied:
Nice, thanks. You may hate me for pointing this out, but many of the tests also set this in their stepTemplates, and I don't think that is needed anymore with your latest. Do you want me to open threads for them all or do you want to just grep like `HOME` in `tasks/managed/*/tests/*` or something?

scoheb replied:
all good. I'll fix them!

johnbieren replied:
idk how you did that so fast

johnbieren commented on stepactions/create-trusted-artifact/create-trusted-artifact.yaml:
Would it make sense to combine create and patch into one stepaction and skip + use into another? So we have 2 instead of 4? It seems the two pairs are together wherever added

scoheb replied:
I thought of that ... but you can only have a single step in a StepAction...`create` is an image/args type (build-trusted-artifacts) step while `patch` is a bash type step.

johnbieren replied:
ah

johnbieren commented on tasks/managed/check-data-keys/check-data-keys.yaml:
```diff
@@ -16,18 +16,109 @@ spec:
+      description: The location where data will be stored
+      type: string
+      default: $(workspaces.data.path)
+    - name: stepActionGitUrl
```
missing from README

scoheb replied:
fixed in all converted tasks. thanks

johnbieren commented on tasks/managed/check-data-keys/check-data-keys.yaml:
```diff
@@ -38,10 +129,11 @@ spec:
             echo "No data JSON was provided."
             exit 1
         fi
+        cat "$(params.dataDir)/$(params.dataPath)"
```
debug leftover?

scoheb replied:
removed

johnbieren commented on tasks/managed/check-data-keys/check-data-keys.yaml:
```diff
@@ -52,7 +144,42 @@ spec:
+        mv "/tmp/systems" "$(params.dataDir)/$(params.dataPath)"
 
-        check-jsonschema --output-format=text --schemafile "/tmp/schema"  "$(workspaces.data.path)/$(params.dataPath)"
+        cat "$(params.dataDir)/$(params.dataPath)"
```
debug leftover?

scoheb replied:
removed

scoheb committed:
feat(RELEASE-1387): trusted artifacts - pr fixes

- docker config json export
- set HOME env var for trusted artifact StepActions
- remove test-embargo-check-rh-issue-wrong-component.yaml

Signed-off-by: Scott Hebert <scoheb@gmail.com>

davidmogar commented on .github/scripts/deploy_registry.sh:
```diff
@@ -0,0 +1,77 @@
+
+deploy_registry() {
+  kubectl apply -k "${script_path}/../resources/registry"
+  retry "kubectl wait --for=condition=Ready --timeout=240s -n kind-registry -l run=registry pod" \
```
Shouldn't be a sleep here as well?

scoheb replied:
not needed...since it seems to work without. Nonetheless, I added a `sleep`

davidmogar commented on pipelines/managed/push-artifacts-to-cdn/README.md:
```diff
@@ -20,3 +20,9 @@ It uses InternalRequests so that it can be run on both public and private cluste
+## Changes in 0.2.0
+* Update all tasks that now support trusted artifacts to specify the stepAction* parameters
+
+## Changes in 0.1.0
```
Why are there two different versions in the changelog if the task goes from 0.0.1 to 0.2.0? I've seen it in others.

scoheb replied:
they were incremental changes that were supposed to land before this large, major PR got merged.

davidmogar replied:
Shouldn't then be removed or put together? It's ok to me if you want to leave it like this.

scoheb replied:
yes. in my latest rebase, I cleaned all the pipelines up to put them together

davidmogar commented on stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations.yaml:
```diff
@@ -0,0 +1,37 @@
+    set -eo pipefail
+    
+    if [ "${ociStorage:?}" == "empty" ]; then
+      echo "oci storage not detected via ociStorage ... skipping trusted artifacts tasks"
```
Nitpick...
```suggestion
      echo "oci storage not detected via ociStorage... skipping trusted artifacts tasks"
```

scoheb replied:
fixed my picky friend

scoheb commented:
/retest

scoheb committed:
feat(RELEASE-1387): trusted artifacts - conforma

- use new git sha param for git resolver

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - HOME

- remove HOME env in StepTemplates since it is
  now defined directly in StepActions

Signed-off-by: Scott Hebert <scoheb@gmail.com>

johnbieren commented on tasks/managed/close-advisory-issues/README.md:
```diff
@@ -8,10 +8,15 @@ Issues in other servers will be skipped without the task failing.
+| dataPath              | Path to data JSON in the data workspace                                                     | No       | -                                                         |
+| advisoryUrl           | The url of the advisory the issues were fixed in. This is added in a comment on the issue   | No       | -                                                         |
+| stepActionGitUrl      | The url to the git repo where the release-service-catalog stepActions to be used are stored | Yes      | https://github.com/konflux-ci/release-service-catalog.git |
+| stepActionGitRevision | The revision in the stepActionGitUrl repo to be used                                        | Yes      | production                                                |
```
Missing ociStorage, ociArtifactExpiresAfter, trustedArtifactsDebug, orasOptions, sourceDataArtifacts, subdirectory, dataDir

scoheb replied:
fixed

johnbieren commented on tasks/managed/collect-atlas-params/README.md:
```diff
@@ -8,6 +8,16 @@ skipped.
-|----------|----------------------------------------------------------------------------------------------------------------|----------|---------------|
-| dataPath | Path to the merged data JSON file generated by collect-data task and containing the Atlas configuration option | No       | -             |
+| Name                    | Description                                                                                                                | Optional | Default value                                             |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------|----------|-----------------------------------------------------------|
```
missing orasOptions and trustedArtifactsDebug

scoheb replied:
fixed

johnbieren commented on tasks/managed/collect-cosign-params/README.md:
```diff
@@ -4,9 +4,19 @@ Tekton task that collects cosign options from the data file.
-|-----------|-------------------------------------------------------------------------|----------|----------------|
-| dataPath  | Path to the JSON string of the merged data to use in the data workspace | No       | -              |
+| Name                    | Description                                                                                                                | Optional | Default value                                             |
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------|----------|-----------------------------------------------------------|
```
missing trustedArtifactsDebug and orasOptions

scoheb replied:
fixed

johnbieren commented on tasks/managed/collect-data/README.md:
```diff
@@ -18,14 +18,21 @@ should not be present in the Release data section).
+| snapshot                | Namespaced name of the Snapshot                                                                                             | No         | -                                                         |
+| subdirectory            | Subdirectory inside the workspace to be used                                                                                | Yes        | ""                                                        |
+| ociStorage              | The OCI repository where the Trusted Artifacts are stored                                                                   | Yes        | empty                                                     |
+| ociArtifactExpiresAfter | Expiration date for the trusted artifacts created in the OCI repository. An empty string means the artifacts do not expir e | Yes        | 1d                                                        |
```
missing orasOptions, trustedArtifactsDebug, dataDir

scoheb replied:
fixed

johnbieren commented on tasks/managed/collect-data/tests/test-collect-data-print-pipeline-ref-nongit.yaml:
```diff
@@ -143,10 +231,11 @@ spec:
               #!/usr/bin/env bash
               set -eux
 
-              touch "$(workspaces.data.path)/mock_curl.txt"
```
this is a weird test.. touch a file then see if it is empty. fail if it isn't :thinking: 
anyways, not introduced in this change, so can resolve without action

johnbieren commented on tasks/managed/collect-pyxis-params/README.md:
```diff
@@ -6,9 +6,19 @@ for downstream tasks to use.
+|-------------------------|----------------------------------------------------------------------------------------------------------------------------|----------|-----------------------------------------------------------|
+| dataPath                | Path to the merged data JSON file generated by collect-data task and containing the pyxis configuration options to use     | No       | -                                                         |
+| ociStorage              | The OCI repository where the Trusted Artifacts are stored                                                                  | Yes      | empty                                                     |
+| ociArtifactExpiresAfter | Expiration date for the trusted artifacts created in the OCI repository. An empty string means the artifacts do not expire | Yes      | 1d                                                        |
```
missing trustedArtifactsDebug and orasOptions

scoheb replied:
fixed

johnbieren commented on tasks/managed/create-advisory/create-advisory.yaml:
```diff
@@ -45,31 +45,119 @@ spec:
+      description: The location where data will be stored
+      type: string
+      default: $(workspaces.data.path)
+    - name: stepActionGitUrl
```
Because this task creates internalRequests, it already has taskGitUrl and taskGitRevision. What about just using those for the step actions resolvers? I think you do this in the pipelines already (they'll need to be removed from the README too if you go for it)

scoheb replied:
I renamed `stepActionGit*` to `taskGit*` to be consistent with those tasks already defining `taskGit*`

johnbieren commented on tasks/managed/create-advisory/README.md:
```diff
@@ -7,18 +7,28 @@ Only all `redhat-pending` or all `redhat-prod` repositories may be specified in
+| taskGitUrl                | The url to the git repo where the release-service-catalog tasks to be used are stored                                      | No       | -                                                         |
+| taskGitRevision           | The revision in the taskGitUrl repo to be used                                                                             | No       | -                                                         |
+| ociStorage                | The OCI repository where the Trusted Artifacts are stored                                                                  | Yes      | empty                                                     |
+| ociArtifactExpiresAfter   | Expiration date for the trusted artifacts created in the OCI repository. An empty string means the artifacts do not expire | Yes      | 1d                                                        |
```
trustedArtifactsDebug and orasOptions

scoheb replied:
fixed

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
```diff
@@ -731,6 +830,10 @@ spec:
           value: $(params.taskGitUrl)
         - name: taskGitRevision
           value: $(params.taskGitRevision)
+        - name: stepActionGitUrl
```
this would have to be removed too

scoheb replied:
fixed

johnbieren reviewed (commented):
Before I go and comment on all of them, how about having the tasks that create InternalRequests reusing the taskGit* params for the step actions?

johnbieren commented on tasks/managed/create-pyxis-image/README.md:
```diff
@@ -9,15 +9,25 @@ The relative path of the pyxis.json file in the data workspace is output as a ta
+| dataPath                | Path to the JSON string of the merged data to use in the data workspace                                                                                                                                                                                                                                                                                                             ...
+| ociStorage              | The OCI repository where the Trusted Artifacts are stored                                                                                                                                                                                                                                                                                                                           ...
+| ociArtifactExpiresAfter | Expiration date for the trusted artifacts created in the OCI repository. An empty string means the artifacts do not expire                                                                                                                                                                                                                                                          ...
+| sourceDataArtifact      | Location of trusted artifacts to be used to populate data directory                                                                                                                                                                                                                                                                                                                 ...
```
trustedArtifactsDebug and orasOptions

scoheb replied:
fixed

johnbieren commented on stepactions/create-trusted-artifact/create-trusted-artifact.yaml:
```diff
@@ -0,0 +1,32 @@
+      default: "empty"
+    - name: dataDir
+      type: string
+    - name: subdirectory
```
Is there any reason to not include this in dataDir? And save needing this parameter

scoheb replied:
fixed

johnbieren commented on stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations.yaml:
```diff
@@ -0,0 +1,37 @@
+    - name: ociStorage
+      type: string
+      default: "empty"
+    - name: subdirectory
```
Is there any reason to not include this in dataDir? And save needing this parameter

scoheb replied:
fixed

scoheb committed:
feat(RELEASE-1387): trusted artifacts - collector

- update tests for collectors

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - readme

- update params in README files

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - params

- re-use taskGitUrl and taskGitRevision parameters

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - stepaction

- combine stepaction params dataDir and subdirectory

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - tests

- default taskGit param values for tests

Signed-off-by: Scott Hebert <scoheb@gmail.com>

jinqi7 commented:
/retest

scoheb commented:
> Before I go and comment on all of them, how about having the tasks that create InternalRequests reusing the taskGit* params for the step actions?

I renamed `stepActionGit*` to `taskGit*` to be consistent with those tasks already defining `taskGit*`

johnbieren commented on pipelines/internal/check-embargoed-cves/README.md:
Are the pipelines/internal README changes intentional? If yes, I am not too sure why, but you'd also have to update the yamls

scoheb replied:
not intentional. reverted the README changes in the /internal/

johnbieren commented on pipelines/managed/e2e/README.md:
```diff
@@ -18,13 +18,16 @@ affected by RHTAP services or which results could affect the RHTAP workflow.
 * Add new parameter `verify_ec_task_git_revision` needed for consuming the verify-conforma task
   via git resolver
 
+## Changes in 3.2.0
```
I think rebase went haywire. This should be above the other changelog entry and it should go to 3.3.0

scoheb replied:
fixed

johnbieren commented on pipelines/managed/e2e/README.md:
```diff
@@ -18,13 +18,16 @@ affected by RHTAP services or which results could affect the RHTAP workflow.
 | verify_ec_task_bundle           | The location of the bundle containing the verify-enterprise-contract task                                                          | No       | -                                                         |
 | verify_ec_task_git_revision     | The git revision to be used when consuming the verify-conforma task                                                                | No       | -                                                         |
-| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                              | Yes      | https://github.com/konflux-ci/release-service-catalog.git |
+| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                              | No        | "" |
```
I think you can leave this line as is, but if you do want to remove the default, you'll have to update the yaml too (applies throughout the pipelines)

scoheb replied:
I reverted the changes to the README files. So the default stays in both READMEs and yamls

johnbieren replied:
I think you only did that in the `internal` READMEs (this one for example still has the change)

scoheb replied:
READMEs should be fixed

johnbieren replied:
nope

johnbieren commented on pipelines/managed/push-disk-images-to-cdn/README.md:
```diff
@@ -17,14 +17,18 @@ Tekton Pipeline to push disk images to a cdn using pulp
   via git resolver
 
-## Changes in 1.5.0
+## Changes in 1.5.1
```
think this is a rebase error

scoheb replied:
fixed

johnbieren commented on pipelines/managed/e2e/README.md:
```diff
@@ -18,13 +18,16 @@ affected by RHTAP services or which results could affect the RHTAP workflow.
   via git resolver
 
+## Changes in 3.2.0
+* Update all tasks that now support trusted artifacts to specify the stepAction* parameters
```
This is also a nit pick, fine to leave as is IMO, but you could like `s/specify the stepAction* parameters/specify the taskGit* parameters for the step action resolvers/g` the READMEs

scoheb replied:
fixed

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
```diff
@@ -599,6 +678,10 @@ spec:
           value: $(params.taskGitUrl)
         - name: taskGitRevision
           value: $(params.taskGitRevision)
+        - name: taskGitUrl
```
this task already has these. I think it will cause an error to pass them twice

johnbieren commented on pipelines/managed/rh-advisories/rh-advisories.yaml:
```diff
@@ -731,6 +830,10 @@ spec:
           value: $(params.taskGitUrl)
         - name: taskGitRevision
           value: $(params.taskGitRevision)
+        - name: taskGitUrl
```
already defined

johnbieren commented on pipelines/managed/rh-push-to-external-registry/rh-push-to-external-registry.yaml:
```diff
@@ -377,6 +413,10 @@ spec:
           value: $(params.taskGitUrl)
         - name: taskGitRevision
           value: $(params.taskGitRevision)
+        - name: taskGitUrl
```
already there

johnbieren commented on pipelines/managed/rh-push-to-registry-redhat-io/rh-push-to-registry-redhat-io.yaml:
```diff
@@ -491,6 +539,10 @@ spec:
           value: $(params.taskGitUrl)
         - name: taskGitRevision
           value: $(params.taskGitRevision)
+        - name: taskGitUrl
```
already there

johnbieren commented on tasks/managed/apply-mapping/apply-mapping.yaml:
```diff
@@ -22,22 +22,104 @@ spec:
+      description: Subdirectory inside the workspace to be used
+      type: string
+      default: ""
+    - name: dataDir
```
Now that the stepactions combine dataDir and subdirectory, can't we do the same thing in the tasks? I see
```
        - name: workDir
          value: $(params.dataDir)/$(params.subdirectory)
```
which makes me think probably?

johnbieren replied:
(The stepactions are definitely cleaner with them combined IMO, but I was hoping we would be able to not have separate params in the tasks too)

scoheb replied:
I have managed to remove as many subdirectory params as I could. A couple of tasks still needed the parameter to help when running unit tests.

jinqi7 commented:
/retest

scoheb committed:
feat(RELEASE-1387): trusted artifacts - fix rebase

- run collector test

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - pipeline

- fix duplicate params for taskGit*

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - readmes

- fix overzealous replacements

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - subdir

- remove subdirectory parameters from most tasks
- to avoid tar extraction errors, add `release` subdur
  to `dataDir`

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - conforma

- rework passing of snapshot path

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - readmes 2

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - test

- add subdir to dataDir parameter in tests

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - rhio

- update rhio pipeline for changes in subdirectory
  parameter

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): trusted artifacts - subdir

- fix update-component-sbom

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): use trusted artifacts - lint

- fix linting errors

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): use trusted artifacts - update

- fix update-cr-status

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): use trusted artifacts - rhio

- revert change to conforma git params

Signed-off-by: Scott Hebert <scoheb@gmail.com>

scoheb committed:
feat(RELEASE-1387): ta - apply-mapping

- revert removal of default for dataDir

Signed-off-by: Scott Hebert <scoheb@gmail.com>

johnbieren commented on tasks/managed/close-advisory-issues/close-advisory-issues.yaml:
```diff
@@ -104,3 +183,35 @@ spec:
         done
 
         exit $RC
+    - name: create-trusted-artifact
```
I think this task just reads from the data file and doesn't modify it or create any new things. Probably can remove this (and other related stuff)?

scoheb replied:
the artifacts are needed for check-result in the Task's unit tests.

johnbieren commented on tasks/managed/collect-data/README.md:
```diff
@@ -18,14 +18,25 @@ should not be present in the Release data section).
+| taskGitUrl              | The url to the git repo where the release-service-catalog tasks and stepactions to be used are stored                      | Yes       | https://github.com/konflux-ci/release-service-catalog.git |
+| taskGitRevision         | The revision in the taskGitUrl repo to be used                                                                             | Yes       | production                                                |
+
+## Changes in 5.0.0
```
Already has a 5.0.0

scoheb replied:
fixed

johnbieren commented on tasks/managed/collect-data/collect-data.yaml:
```diff
@@ -90,40 +137,51 @@ spec:
-          mkdir -p "$(workspaces.data.path)/$(params.subdirectory)"
+          mkdir -p "$(params.dataDir)/$(params.subdirectory)"
+          if [ "$(params.ociStorage)" == "empty" ]; then
+            echo "oci storage not detected via params.ociStorage ... skipping trusted artifacts tasks"
```
is there a reason why not to use the stepaction for this or was it just forgotten?

scoheb replied:
possibly forgotten. I replaced this with a call to the StepAction.

johnbieren commented on tasks/managed/create-product-sbom/README.md:
```diff
@@ -5,9 +5,21 @@ releaseNotes content.
+| trustedArtifactsDebug   | Flag to enable debug logging in trusted artifacts. Set to a non-empty string to enable                                     | Yes        | ""                      |
+| orasOptions             | oras options to pass to Trusted Artifacts calls                                                                            | Yes        | ""                      | 
+| sourceDataArtifact      | Location of trusted artifacts to be used to populate data directory                                                        | Yes        | ""                      |
+| subdirectory            | Subdirectory inside the workspace to be used                                                                               | Yes        | ""                      |
```
this can be removed, it isn't in the yaml

scoheb replied:
fixed

johnbieren commented on tasks/managed/populate-release-notes/populate-release-notes.yaml:
```diff
@@ -147,11 +225,12 @@ spec:
             echo "No data JSON was provided."
             exit 1
         fi
+        cat "$(params.dataDir)/$(params.dataPath)"
```
debug leftover

scoheb replied:
fixed

johnbieren commented on tasks/managed/populate-release-notes/populate-release-notes.yaml:
```diff
@@ -163,6 +242,7 @@ spec:
         if [ "$(jq -r '.releaseNotes.type' "${DATA_FILE}")" != "RHSA" ] ; then
             echo "Type is not RHSA. Ensuring references key exists, but not adding any"
             jq '.releaseNotes.references += []' "${DATA_FILE}" > /tmp/data.tmp && mv /tmp/data.tmp "${DATA_FILE}"
+            cat "$(params.dataDir)/$(params.dataPath)"
```
debug leftover

johnbieren replied:
@scoheb missed one

johnbieren commented on tasks/managed/populate-release-notes/populate-release-notes.yaml:
```diff
@@ -180,3 +260,36 @@ spec:
 
         # Remove duplicate references
         jq '.releaseNotes.references |= unique' "${DATA_FILE}" > /tmp/data.tmp && mv /tmp/data.tmp "${DATA_FILE}"
+        cat "$(params.dataDir)/$(params.dataPath)"
```
debug leftover

johnbieren commented on tasks/managed/rh-sign-image-cosign/rh-sign-image-cosign.yaml:
```diff
@@ -32,10 +32,88 @@ spec:
   workspaces:
     - name: data
       description: Workspace to read and save files
+  results:
```
I don't have the best understanding of this task, but I don't think it writes to the data workspace at all

johnbieren commented on tasks/managed/rh-sign-image/rh-sign-image.yaml:
```diff
@@ -55,10 +49,88 @@ spec:
   workspaces:
     - name: data
       description: workspace to read and save files
+  results:
```
I don't think this one writes to the data workspace

johnbieren commented on tasks/managed/run-file-updates/run-file-updates.yaml:
```diff
@@ -40,39 +40,129 @@ spec:
+                  "$0: ERROR '$command' failed at line $line - exited with status $err"
+                touch "$(params.dataDir)/error"
+            fi
+            exit 0
```
This isn't an internal task. Why are we making it never fail?

johnbieren commented on tasks/managed/run-file-updates/run-file-updates.yaml:
```diff
@@ -40,39 +40,129 @@ spec:
+          value: $(params.sourceDataArtifact)
     - name: run-script
       image: quay.io/konflux-ci/release-service-utils:0f82be4be43294b6a96846d87ef7f7c0b9e34267
+      onError: continue
```
why?

scoheb replied:
since we now don't have a persistent workspace, we want to be able to inspect the workspace after at least 1 of the file-updates has failed. To do that, we need to create the artifacts before we exit with a failure.

If we mark the test as expected to fail, without having the `onError: continue`, we never get to the check-result step to examine.

johnbieren replied:
> If we mark the test as expected to fail, without having the onError: continue, we never get to the check-result step to examine.

This isn't a test, it is the task yaml. What are you talking about check-result?

scoheb replied:
yes...but we need to be able to provide artifacts so that we can test the task after it has completed.

I am trying to cover the use case for `multiple-ir-first-fails`

otherwise, in the case for multiple `file-updates`...one of which may fail, we'd have to blindly resort to marking the test as expected to fail and count the failures hoping it is the one that has failed is the one we are expecting to fail.

if you prefer, I can remove this `onError: continue`, and simplify the tests.

johnbieren replied:
From that test
```
              # Fetch all the InternalRequest resources
              allRequests=$(kubectl get InternalRequest -o json)

              # Count the number of InternalRequests
              requestsCount=$(echo "$allRequests" | jq -r '.items | length')

              # Check if the number of InternalRequests is as expected
              if [ "$requestsCount" -ne "1" ]; then
                echo "Unexpected number of InternalRequests. Expected: 1, Found: $requestsCount"
                exit 1
              fi
```
so there aren't even multiple IRs in that test

scoheb replied:
there is clearly a bug in the test...I am re-working it.

scoheb replied:
there was no bug in the tests.

There was an incorrect assumption that we needed to capture the workspace via TA when a test failed. Turns out that with a few changes in `mocks.sh`, we can have our tests checked without anything from the workspace.

With that, I have also removed the `onError: continue` from the **Task**

johnbieren commented on tasks/managed/run-file-updates/run-file-updates.yaml:
```diff
@@ -40,39 +40,129 @@ spec:
+        # due to set -e, this catches all EXIT and ERR calls and the task should never fail with nonzero exit code
+        trap 'exitfunc $? $LINENO "$BASH_COMMAND"' EXIT
+
+        rm -rf "$(params.dataDir)/error"
```
not sure what this is for

scoheb replied:
see above comment.

johnbieren commented on tasks/managed/update-cr-status/update-cr-status.yaml:
```diff
@@ -29,26 +29,106 @@ spec:
         Path to the directory containing the result files in the data workspace which will be added to the
         resource's status
       type: string
+    - name: resultsDirName
```
This isn't in the README. We need this when we already have resultsDirPath?

scoheb replied:
I removed `resultsDirPath` and replaced it with `resultsDirName`. updated README.

scoheb replied:
Actually, I reverted again to keep the original  `resultsDirPath` and updated README again.

johnbieren commented on tasks/managed/upload-sbom-to-atlas/upload-sbom-to-atlas.yaml:
```diff
@@ -50,24 +50,107 @@ spec:
+      description: The revision in the taskGitUrl repo to be used
   workspaces:
     - name: data
+  results:
```
not sure this one writes to the workspace

scoheb replied:
the artifacts are needed for check-result in the Task's unit tests.

johnbieren commented on tasks/managed/publish-pyxis-repository/tests/test-publish-pyxis-repository-container-multiple-components.yaml:
```diff
@@ -55,56 +92,136 @@ spec:
+              - name: ociStorage
+                value: $(params.ociStorage)
+              - name: workDir
+                value: $(params.dataDir)/
```
This task's tests are the only ones you added the trailing `/` on this line :wink:

scoheb replied:
fixed

scoheb committed:
feat(RELEASE-1387): ta - pr fixes

Signed-off-by: Scott Hebert <scoheb@gmail.com>

johnbieren commented on tasks/managed/publish-pyxis-repository/tests/test-publish-pyxis-repository-source-container-true.yaml:
```diff
@@ -47,46 +84,126 @@ spec:
+              - name: ociStorage
+                value: $(params.ociStorage)
+              - name: workDir
+                value: $(params.dataDir)/
```
trailing `/` :wink:

scoheb replied:
fixed

johnbieren commented on tasks/managed/publish-pyxis-repository/tests/test-publish-pyxis-repository-marked-as-not-pub-on-push.yaml:
```diff
@@ -37,50 +74,130 @@ spec:
+              - name: ociStorage
+                value: $(params.ociStorage)
+              - name: workDir
+                value: $(params.dataDir)/
```
trailing `/` :wink:

scoheb replied:
fixed

johnbieren commented on tasks/managed/publish-pyxis-repository/tests/test-publish-pyxis-repository-no-terms-required.yaml:
```diff
@@ -73,22 +150,61 @@ spec:
+              - name: ociStorage
+                value: $(params.ociStorage)
+              - name: workDir
+                value: $(params.dataDir)/
```
trailing `/` :wink:

scoheb replied:
fixed

johnbieren commented on tasks/managed/publish-pyxis-repository/tests/test-publish-pyxis-repository-skip-publishing.yaml:
```diff
@@ -37,47 +74,127 @@ spec:
+              - name: ociStorage
+                value: $(params.ociStorage)
+              - name: workDir
+                value: $(params.dataDir)/
```
trailing `/` :wink:

scoheb replied:
fixed

johnbieren commented on tasks/managed/publish-pyxis-repository/tests/test-publish-pyxis-repository-source-container-false.yaml:
```diff
@@ -46,48 +83,128 @@ spec:
+              - name: ociStorage
+                value: $(params.ociStorage)
+              - name: workDir
+                value: $(params.dataDir)/
```
trailing `/` :wink:

scoheb replied:
fixed

johnbieren commented on tasks/managed/update-cr-status/update-cr-status.yaml:
```diff
@@ -24,31 +24,106 @@ spec:
+          value: $(params.ociStorage)
+        - name: workDir
+          value: $(params.dataDir)
+    - name: use-trusted-artifact-array
```
what is this doing? Also `params.resultArtifacts` isn't in the README

scoheb replied:
This step tasks the `array-type` parameter and extract the artifacts.

it is the sister of the StepAction `use-trusted-artifact`

scoheb replied:
fixed README too

johnbieren replied:
should it be in the stepactions dir and called via resolver then? I get it is the only task using this, but maybe better for consistency?

scoheb replied:
updated. see note below about `patch`

johnbieren replied:
thank you for linking the issue

jinqi7 commented:
/retest

johnbieren commented on pipelines/managed/push-binaries-to-dev-portal/README.md:
```diff
@@ -17,9 +17,12 @@ Tekton pipeline to sign and release Red Hat binaries to the Red Hat Developer Po
 | verify_ec_task_bundle           | The location of the bundle containing the verify-enterprise-contract task                                                          | No       | -                                                         |
 | verify_ec_task_git_revision     | The git revision to be used when consuming the verify-conforma task                                                                | No       | -                                                         |
-| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                              | Yes      | https://github.com/konflux-ci/release-service-catalog.git |
+| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                              | No        | "" |
```
taskGitUrl  change shouldn't be here

scoheb replied:
fixed

johnbieren commented on pipelines/managed/push-artifacts-to-cdn/README.md:
```diff
@@ -18,9 +18,13 @@ It uses InternalRequests so that it can be run on both public and private cluste
 | verify_ec_task_bundle           | The location of the bundle containing the verify-enterprise-contract task                                                           | No       | -                                                         |
 | verify_ec_task_git_revision     | The git revision to be used when consuming the verify-conforma task                                                                 | No       | -                                                         |
-| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                               | Yes      | https://github.com/konflux-ci/release-service-catalog.git |
+| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                               | No        | "" |
```
line shouldn't be changed

scoheb replied:
fixed

johnbieren commented on pipelines/managed/fbc-release/README.md:
```diff
@@ -17,9 +17,12 @@ Tekton release pipeline to interact with FBC Pipeline
 | verify_ec_task_git_revision     | The git revision to be used when consuming the verify-conforma task                                                                        | No       | -                                                         |
 | postCleanUp                     | Cleans up workspace after finishing executing the pipeline                                                                                 | Yes      | true                                                      |
-| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                                      | Yes      | https://github.com/konflux-ci/release-service-catalog.git |
+| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                                      | No        | "" |
```
line shouldn't be changed

scoheb replied:
fixed

johnbieren commented on pipelines/managed/e2e/README.md:
```diff
@@ -18,9 +18,12 @@ affected by RHTAP services or which results could affect the RHTAP workflow.
 | verify_ec_task_bundle           | The location of the bundle containing the verify-enterprise-contract task                                                          | No       | -                                                         |
 | verify_ec_task_git_revision     | The git revision to be used when consuming the verify-conforma task                                                                | No       | -                                                         |
-| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                              | Yes      | https://github.com/konflux-ci/release-service-catalog.git |
+| taskGitUrl                      | The url to the git repo where the release-service-catalog tasks to be used are stored                                              | No        | "" |
```
line shouldn't be changed

scoheb replied:
fixed

scoheb commented:
/retest

scoheb commented:
/retest

johnbieren commented on tasks/managed/run-file-updates/tests/test-run-file-updates-multiple-ir-first-fails.yaml:
```diff
@@ -6,21 +6,59 @@ metadata:
+      type: string
   tasks:
     - name: setup
+      params:
```
typo? doesn't seem like you added any params to this task

scoheb replied:
fixed

johnbieren commented on tasks/managed/run-file-updates/tests/test-run-file-updates-multiple-ir-last-fails.yaml:
```diff
@@ -10,17 +10,56 @@ spec:
+      type: string
   tasks:
     - name: setup
+      params:
```
typo?

scoheb replied:
fixed

johnbieren commented on tasks/managed/run-file-updates/tests/test-run-file-updates.yaml:
```diff
@@ -8,17 +8,56 @@ spec:
+      type: string
   tasks:
     - name: setup
+      params:
```
typo?

scoheb replied:
fixed

johnbieren commented on tasks/managed/update-cr-status/update-cr-status.yaml:
```diff
@@ -29,18 +29,87 @@ spec:
+      description: oras options to pass to Trusted Artifacts calls
+      type: string
+      default: ""
+    - name: resultArtifacts
```
This isn't in the README

scoheb replied:
fixed

johnbieren commented on tasks/managed/update-cr-status/update-cr-status.yaml:
```diff
@@ -49,6 +118,21 @@ spec:
             # are merged (only arrays are not merged properly with this notation).
             RESULTS_JSON=$(jq -c "${RESULTS_JSON} * ." "${resultsFile}")
+
+        #        resultPaths=$(find "$(params.dataDir)" -name "$(params.resultsDirPath)" -type d)
```
debug leftover?

scoheb replied:
removed

johnbieren commented on tasks/managed/update-cr-status/tests/test-update-cr-status-multiple-result-files.yaml:
```diff
@@ -53,6 +94,37 @@ spec:
                 releasePlan: foo
               EOF
               kubectl apply -f release
+          - name: skip-trusted-artifact-operations-array
```
maybe we can have `stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations.yaml` and `stepactions/skip-trusted-artifact-operations/skip-trusted-artifact-operations-array.yaml`?

scoheb replied:
I was able to convert `create-trusted-artifact-array` and `use-trusted-artifact-array`.

However, `patch-source-data-artifact-result-array` cannot be converted due to https://github.com/tektoncd/pipeline/issues/2374

johnbieren replied:
thanks for linking the issue in the yaml

scoheb commented:
/retest

scoheb commented:
/retest

scoheb commented:
e2e showed:

```
step-clone

INFO: Using mounted CA bundle: /mnt/trusted-ca/ca-bundle.crt
{"level":"info","ts":1744741181.6275237,"caller":"git/git.go:176","msg":"Successfully cloned https://github.com/redhat-appstudio-qe/fbc-sample-repo-test @ ccc3571b07b84163028fbc674f26dc6f695fb7e5 (grafted, HEAD) in path /var/workdir/source"}
{"level":"info","ts":1744741181.6603377,"caller":"git/git.go:215","msg":"Successfully initialized and updated submodules in path /var/workdir/source"}
Merge option disabled. Using checked-out revision ccc3571b07b84163028fbc674f26dc6f695fb7e5 directly.
step-symlink-check

Running symlink check
step-create-trusted-artifact

Prepared artifact from /var/workdir/source (sha256:93e8ba83302dc66425f2011e7588dce209e01a57193ee6b85f4b3169282bcda2)
Using token for quay.io
Error response from registry: recognizable error message not found: HEAD "https://quay.io/v2/redhat-user-workloads-stage/dev-release-team-tenant/test-component-pac-ihvkyl/manifests/sha256:db867ecb51e9e5520518918d71feac739823df6e1a78a4736a9e0addf63c4968": response status code 401: Unauthorized
Command exited with non-zero status 1
```

scoheb commented:
/retest
